File: Documentation/5.4/Samples/python/ClientApi/Operations/Maintenance/Indexes/ResumeIndex.py
from ravendb import StartIndexOperation, GetIndexingStatusOperation, StopIndexingOperation
from ravendb.documents.indexes.definitions import IndexRunningStatus
from ravendb.documents.operations.definitions import VoidMaintenanceOperation

from examples_base import ExampleBase


class Foo:
    # region syntax
    class StartIndexOperation(VoidMaintenanceOperation):
        def __init__(self, index_name: str): ...

    # endregion


class ResumeIndex(ExampleBase):
    def setUp(self):
        super().setUp()

    def test_resume_index(self):
        with self.embedded_server.get_document_store("ResumeIndex") as store:
            self.add_index_orders_totals(store)
            store.maintenance.send(StopIndexingOperation())
            # region resume_index
            # Define the resume index operation, pass the index name
            resume_index_op = StartIndexOperation("Orders/Totals")

            # Execute the operation by passing it to maintenance.send
            store.maintenance.send(resume_index_op)

            # At this point:
            # Index 'Orders/Totals' is resumed on the preferred node

            # Can verify the index status on the preferred node by sending GetIndexingStatusOperation
            indexing_status = store.maintenance.send(GetIndexingStatusOperation())

            index = [x for x in indexing_status.indexes if x.name == "Orders/Totals"][0]
            self.assertEqual(index.status, IndexRunningStatus.RUNNING)
            # endregion

File: Documentation/5.4/Samples/php/ClientApi/Operations/Maintenance/Indexes/ResumeIndex.php
<?php

namespace RavenDB\Samples\ClientApi\Operations\Maintenance\Indexes;

use PHPUnit\Framework\TestCase;
use RavenDB\Documents\DocumentStore;
use RavenDB\Documents\Indexes\IndexingStatus;
use RavenDB\Documents\Operations\Indexes\GetIndexingStatusOperation;
use RavenDB\Documents\Operations\Indexes\StartIndexOperation;

class ResumeIndex extends TestCase
{
    public function samples(): void
    {
        $store = new DocumentStore();
        try {
            # region resume_index
            // Define the resume index operation, pass the index name
            $resumeIndexOp = new StartIndexOperation("Orders/Totals");

            // Execute the operation by passing it to Maintenance.Send
            $store->maintenance()->send($resumeIndexOp);

            // At this point:
            // Index 'Orders/Totals' is resumed on the preferred node

            // Can verify the index status on the preferred node by sending GetIndexingStatusOperation
            /** @var IndexingStatus $indexingStatus */
            $indexingStatus = $store->maintenance()->send(new GetIndexingStatusOperation());

            $indexes = array_filter($indexingStatus->getIndexes()->getArrayCopy(), function ($v, $k) {
                return $v->getName() == "Orders/Totals";
            });
            /** @var IndexingStatus $index */
            $index = $indexes[0];

            $this->assertTrue($index->getStatus()->isRunning());
            # endregion
        } finally {
            $store->close();
        }
    }
}

/*
interface IFoo
{
    # region syntax
    // class name begins with "Start" but this is still the "Resume" operation
    StartIndexOperation(?string $indexName)
    # endregion
}
*/
